[Enhancement] Add memory limit for ByteBuffer allocation in data ingestion

srlch opened the pull request:
## Why I'm doing:
ByteBuffer may allocate huge chunk of memory and cause BE OOM in this case.

## What I'm doing:
limit the memory allocation using the current thread tracker.

Fixes #issue
https://github.com/StarRocks/StarRocksTest/issues/8102

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

trueeyu commented on be/src/util/byte_buffer.h:
```diff
@@ -61,25 +63,26 @@ struct MemTrackerDeleter {
+        // set memory tracker for this thread
+        SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(tracker);
+        // check limit before allocation
+        TRY_CATCH_BAD_ALLOC(ByteBufferPtr ptr(new ByteBuffer(size), MemTrackerDeleter(tracker)); return ptr;);
```
free load mem_tracker?

srlch committed:
[Enhancement] Add memory limit for ByteBuffer allocation in data ingestion

Why I'm doing:
ByteBuffer may allocate huge chunk of memory and cause BE OOM in this case.

What I'm doing:
limit the memory allocation using the current thread tracker.

Signed-off-by: srlch <linzichao@starrocks.com>

srlch committed:
fix format

Signed-off-by: srlch <linzichao@starrocks.com>

srlch committed:
fix

Signed-off-by: srlch <linzichao@starrocks.com>

trueeyu commented on be/src/http/action/transaction_stream_load.cpp:
```diff
@@ -559,7 +569,12 @@ void TransactionStreamLoadAction::on_chunk_data(HttpRequest* req) {
                 }
 
-                ctx->buffer = ByteBuffer::allocate_with_tracker(std::max(len, ctx->kDefaultBufferSize));
+                auto res = ByteBuffer::allocate_with_tracker(std::max(len, ctx->kDefaultBufferSize));
```
This code has been written many times, can it be encapsulated?

srlch committed:
refactor

Signed-off-by: srlch <linzichao@starrocks.com>

srlch committed:
fix

Signed-off-by: srlch <linzichao@starrocks.com>
